src/main/java/hudson/plugins/ec2/SlaveTemplate.java

```diff
@@ -588,6 +590,10 @@ private EC2AbstractSlave provisionOndemand(TaskListener listener, Label required
                 if (StringUtils.isNotBlank(getIamInstanceProfile())) {
                     riRequest.setIamInstanceProfile(new IamInstanceProfileSpecification().withArn(getIamInstanceProfile()));
                 }
+                // For T2 instances, set CPU credit type accordingly
+                if (type.toString().toLowerCase().startsWith("t2") && t2Unlimited) {
```

Comment:
I would switch the order:
`if (t2Unlimited && type.toString().toLowerCase().startsWith("t2")) {`
If **`t2Unlimited`** is false, it doesn't matter what the value of **`type`** is and will save on evaluating the toString operation.